src/core/dataframe.js

```diff
@@ -61,26 +61,46 @@ export default class Dataframe {
         gl.bufferData(gl.ARRAY_BUFFER, ids, gl.STATIC_DRAW);
     }
 
-    setStyle(style) {
-        this.style = style;
-    }
-
     getFeaturesAtPosition(pos, style) {
         switch (this.type) {
             case 'point':
-                return this._getPointsAtPosition(pos);
+                return this._getPointsAtPosition(pos, style);
             case 'line':
                 return this._getLinesAtPosition(pos, style);
             case 'polygon':
-                return this._getPolygonAtPosition(pos);
+                return this._getPolygonAtPosition(pos, style);
             default:
                 return [];
         }
     }
 
-    _getPointsAtPosition(pos) {
-        console.log(pos);
-        return [];
+    _getPointsAtPosition(p, style) {
+        p = wToR(p.x, p.y, { center: this.center, scale: this.scale });
+        const points = this.decodedGeom.vertices;
+        const features = [];
+        const widthScale = 1 / this.renderer.gl.canvas.height / this.scale * this.renderer._zoom;
+        const columnNames = Object.keys(this.properties);
+        const styleWidth = style.getWidth();
+        const styleStrokeWidth = style.getStrokeWidth();
+        for (let i = 0; i < points.length; i += 2) {
+            const featureID = i / 2;
+            const center = {
+                x: points[i],
+                y: points[i + 1],
+            };
+            const f = {};
+            columnNames.forEach(name => {
+                f[name] = this.properties[name][featureID];
+            });
+            const scale = (styleWidth.eval(f) + styleStrokeWidth.eval(f)) * widthScale;
```

Comment:
I think we should use `(styleWidth.eval(f) + styleStrokeWidth.eval(f)) * widthScale * 0.5` here

Reply:
Why? have you manually tried the example? This matched the rendering in my tests

Reply:
Yep, so I am missing something. This is the model, right?
![point](https://user-images.githubusercontent.com/2227572/37962621-01cf9ea4-31bc-11e8-8061-8377a1a707da.png)

So may be the 0.5 is implicit in widthScale.
Could you spot the missing part (just for my clarification :wink:)

Reply:
I see your point, I think it's due to `1 / this.renderer.gl.canvas.height`. The vertex shader converts the coordinates to the [-1,1] range, therefore, a `2 / this.renderer.gl.canvas.height` is what we are doing at render time. Here the `2` is being canceled by the `0.5`.

Reply:
Ok. It would be nice to have a comment with this information in the code to avoid future misunderstandings.